### PR TITLE
Target k3d in integration tests

### PR DESCRIPTION
**Pull request by hassomehide**
**Description**
Follow-up to #4224. Adds integration test jobs that target a k3d cluster.

### CODE REVIEW

File: deploy/skaffold/Dockerfile [hunk 1]
```diff
@@ -13,7 +13,7 @@
 # limitations under the License.
 
 # This base image has to be updated manually after running `make build_deps`
-FROM gcr.io/k8s-skaffold/build_deps:d54e32f303a9bf5ce42978be0b04ae6cba235d37 as builder
+FROM gcr.io/k8s-skaffold/build_deps:8acc60984aef0c23435ed0c3b69d2a19310f4acd as builder
```
Comment (hassomehide) [inline deploy/skaffold/Dockerfile:16]: (this has to be pushed by someone who has access)

**Comment by nkubala** [comment]
@hassomehide I pushed the dep image, and while the kokoro integration tests are passing the k3d partitions are failing on travis. any idea why?

**Comment by hassomehide** [comment]
> @hassomehide I pushed the dep image, and while the kokoro integration tests are passing the k3d partitions are failing on travis. any idea why?

Travis builds are running on docker 18.06 which seems to cause an issue with k3d. The solution right now is to upgrade the docker version in k3d integration test jobs.

**Comment by hassomehide** [comment]
I upgraded the docker version in k3d partitions which seemed to fix the issue. However, k3d v3.0.0-beta.2 has since been released and so the deps image needs to be pushed again @nkubala

**Comment by nkubala** [comment]
@hassomehide I pushed it, every test is passing except one of the travis tests and it's failed twice in a row. any ideas about that one?

**Comment by hassomehide** [comment]
@nkubala I haven't been able to consistently reproduce this yet, but I've seen it happen a few times with different tests like TestDebug/kubectl and TestRun/microservices. Will try to run them in an ubuntu vm to get a more similar environment to the travis builds.

Meanwhile I have updated the deps image with k3d v3.0.0-rc.2, looks like a stable release is due soon.

**Comment by tejal29** [comment]
@hassomehide Restarted your build. The previous failures happened in the early setup stages.

**Comment by dgageot** [comment]
@hassomehide I'm about to push a PR that includes k3d 3.0.0 in the skaffold image. That should help you to rebase and fix this PR

**Comment by hassomehide** [comment]
@dgageot Thanks! Apparently k3d had breaking cli changes between v3.0.0-rc.6 and v3.0.0, so this PR needs some more work, will look into it in a few days...

**Pull request merged**